master-firmware/src/strategy/color_sequence_server.c

```diff
@@ -0,0 +1,67 @@
+#include <ch.h>
+#include <hal.h>
+
+#include <error/error.h>
+
+#include "main.h"
+#include "priorities.h"
+
+#include "robot_helpers/eurobot2018.h"
+#include "color_sequence_server.h"
+
+#define COLOR_SEQUENCE_SERVER_STACKSIZE 512
+
+// Bluetooth UART
+#define BLUETOOTH_UART_BAUDRATE 19200
+static const SerialConfig bt_uart_config = {
+    .speed = BLUETOOTH_UART_BAUDRATE,
+    .cr1 = 0,
+    .cr2 = USART_CR2_STOP1_BITS | USART_CR2_LINEN,
+    .cr3 = 0
+};
+
+static THD_FUNCTION(color_sequence_server_thd, arg) {
+    (void)arg;
+    chRegSetThreadName(__FUNCTION__);
+
+    uint8_t buffer[16];
+    sdStart(&SD2, &bt_uart_config);
+
+    static enum cube_color colors[5] = {
+        CUBE_UNKNOWN, CUBE_UNKNOWN, CUBE_UNKNOWN, CUBE_UNKNOWN, CUBE_UNKNOWN
+    };
+    static messagebus_topic_t colors_topic;
+    static MUTEX_DECL(colors_topic_lock);
+    static CONDVAR_DECL(colors_topic_condvar);
+    messagebus_topic_init(&colors_topic, &colors_topic_lock, &colors_topic_condvar, &colors[0], sizeof(colors));
+    messagebus_advertise_topic(&bus, &colors_topic, "/colors");
+
+    NOTICE("Color sequence server up, listening over BT");
+
+    while (true) {
+        size_t len = sdAsynchronousRead(&SD2, buffer, sizeof(buffer));
```

Comment:
This does not look very robust to character drop, but maybe it works ?

Reply:
Not robust, but it didn't fail yet (tested ~10 games) :grimacing:
I'm happy to change it to something more robust, any suggestions?

Reply:
```cpp
do {
    sdRead(&SD2, buffer, 1);
} while(buffer[0] != '\n');

sdRead(&SD2, buffer, 3);
```
Something like that maybe ? That waits for the sync character before parsing the rest ?

Reply:
looks much better, I'll change it